cirq-core/cirq/neutral_atoms/neutral_atom_gateset.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Optional, TYPE_CHECKING
+
+from cirq import ops, protocols, transformers
+from cirq.protocols.decompose_protocol import DecomposeResult
+
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class NeutralAtomGateset(transformers.CompilationTargetGateset):
+    """A Compilation target intended for neutral atom devices.
+
+    This gateset supports CNOT, CCNOT (TOFFOLI) gates, CZ,
+    CCZ gates, as well as single qubits gates that can be used
+    in a parallel fashion.  The maximum amount of parallelism
+    can be set by arguments.
+
+    Args:
+        max_parallel_z: The maximum amount of parallelism for
+            Z gates.
+        max_parallel_xy: The maximum amount of parallelism for
+            X, Y and PhasedXPow gates.
+    """
+
+    def __init__(self, max_parallel_z: Optional[int] = None, max_parallel_xy: Optional[int] = None):
+        super().__init__(
+            ops.AnyIntegerPowerGateFamily(ops.CNotPowGate),
+            ops.AnyIntegerPowerGateFamily(ops.CCNotPowGate),
+            ops.AnyIntegerPowerGateFamily(ops.CZPowGate),
+            ops.AnyIntegerPowerGateFamily(ops.CCZPowGate),
+            ops.ParallelGateFamily(ops.ZPowGate, max_parallel_allowed=max_parallel_z),
+            ops.ParallelGateFamily(ops.XPowGate, max_parallel_allowed=max_parallel_xy),
+            ops.ParallelGateFamily(ops.YPowGate, max_parallel_allowed=max_parallel_xy),
+            ops.ParallelGateFamily(ops.PhasedXPowGate, max_parallel_allowed=max_parallel_xy),
+            ops.MeasurementGate,
+            ops.IdentityGate,
+            unroll_circuit_op=False,
+        )
+
+    def num_qubits(self) -> int:
+        """Maximum number of qubits on which a gate from this gateset can act upon."""
+        return 2
+
+    def decompose_to_target_gateset(self, op: 'cirq.Operation', moment_idx: int) -> DecomposeResult:
+        """Method to rewrite the given operation using gates from this gateset.
```

Comment:
Add to the docstring that 2q gates are decomposed into CZs because "CZ gates are the highest fidelity 2-qubit gates for neutral atoms." (quoted from docstring of ConvertToNeutralAtomGates optimizer)

Reply:
Done.